pkg/cache/clusterqueue_test.go

```diff
@@ -771,10 +797,69 @@ func TestClusterQueueUpdateWithAdmissionCheck(t *testing.T) {
 			wantReason:  "Terminating",
 			wantMessage: "Can't admit new workloads; clusterQueue is terminating",
 		},
+		{
+			name:     "Active clusterQueue with an AC strategy updated with duplicate single instance AC Controller",
+			cq:       cqWithACStrategy,
+			cqStatus: active,
+			admissionChecks: map[string]AdmissionCheck{
+				"check1": {
+					Active:                       true,
+					Controller:                   "controller1",
+					SingleInstanceInClusterQueue: true,
+				},
+				"check2": {
+					Active:     true,
+					Controller: "controller2",
+				},
+				"check3": {
+					Active:                       true,
+					Controller:                   "controller2",
+					SingleInstanceInClusterQueue: true,
+				},
+			},
+			wantStatus:  pending,
+			wantReason:  "MultipleSingleInstanceControllerAdmissionChecks",
+			wantMessage: `Can't admit new workloads: only one AdmissionCheck of [check2 check3] can be referenced for controller "controller2".`,
+		},
+		{
+			name:     "Active clusterQueue with a FlavorIndependent AC applied per ResourceFlavor",
+			cq:       cqWithACPerFlavor,
+			cqStatus: pending,
+			admissionChecks: map[string]AdmissionCheck{
+				"check1": {
+					Active:            true,
+					Controller:        "controller1",
+					FlavorIndependent: true,
+				},
+			},
+			wantStatus:  pending,
+			wantReason:  "FlavorIndependentAdmissionCheckAppliedPerFlavor",
+			wantMessage: "Can't admit new workloads: AdmissionCheck(s): [check1] cannot be set at flavor level.",
+		},
+		{
+			name:     "Active clusterQueue with a FlavorIndependent MultiKueue AC applied per ResourceFlavor",
+			cq:       cqWithACPerFlavor,
+			cqStatus: pending,
+			admissionChecks: map[string]AdmissionCheck{
+				"check1": {
+					Active:            true,
+					Controller:        kueue.MultiKueueControllerName,
+					FlavorIndependent: true,
+				},
+			},
+			wantStatus:  pending,
+			wantReason:  "FlavorIndependentAdmissionCheckAppliedPerFlavor",
```

Comment:
IIUC this reason should be also specific for MultiKueue as the feature gate is disbaled

Reply:
problematic, bc who this stuff is designed, first reason to appear is actually returned
```go
		return reasons[0], api.TruncateConditionMessage(strings.Join([]string{"Can't admit new workloads: ", strings.Join(messages, ", "), "."}, ""))
```

Reply:
I would think we don't need to generate this reason then, only the dedicated one

Reply:
I will put the new multikueue reason before those old
but if we want to change the overall approach it should be another PR